packages/neptune-extension/src/common/components/upload-modal/UploadModal.tsx

```diff
@@ -92,26 +94,18 @@ const UploadModal: React.FC<UploadModalProps> = ({
     };
 
     if (mode === 'notebook' || metadata.notebookId === undefined) {
-      const notebookMeta = await uploadNotebook(projectId, checkpointMeta, content);
-
-      await platformNotebook.saveNotebookId(notebookMeta.id);
-
-      const notebook = await getNotebook(notebookMeta.id)
-      dispatch(setNotebook(notebook))
+      const notebookMeta = await thunkDispatch(uploadNotebook(projectId, checkpointMeta, content));
+      if (notebookMeta) {
```

Comment:
I feel a bit uneasy with catching an error inside `uploadNotebook` and converting it into a normal application flow (returning undefined), because that would signify that this is a normal mode of operation and not an exception. Ideally we should have one error handler for the whole submit function and any exception should break the flow. Can we think of such solution ? Or at least transform this code block into a function that throws ?

Reply:
IMHO handling exceptions inside actions/reducer is better:
1. I would not like to think each time i dispatch an action, whether I should wrap in try catch or not.
2. I would like to encapsulate all the flow of upload to action, so my components will contain less logic.
3. It gives me a possibility to add add error handler but not in here. I will handle error property from store and present either a Toast or Notice inside modal. It gives me more flexibility

Ideally I would move saveNotebookId and fetchNotebook to action, but that would require rewriting platform into singleton and I dont want to do this in this commit